toolchain/check/context.h

```diff
@@ -367,11 +367,13 @@ class Context {
   // Returns whether the current position in the current block is reachable.
   auto is_current_position_reachable() -> bool;
 
-  // Returns the type ID for a constant of type `type`.
+  // Returns the type ID for a constant is a type value or constraint value,
+  // i.e. it is a value of type `TypeType` or `FacetType`.
   auto GetTypeIdForTypeConstant(SemIR::ConstantId constant_id) -> SemIR::TypeId;
```

Comment:
Hm. We really should stop calling this on facets. I wonder how many places we're still doing that; maybe we could make the comment correct instead?

Reply:
One question, when looking at the typish diagram, I noticed this removes the edge from FacetValue to TypeId but not from FacetType to TypeId, since FacetType has type `TypeType`. Is this the desired outcome for you?